cmd/tempo/app/modules.go

```diff
@@ -413,13 +441,34 @@ func (t *App) setupModuleManager() error {
 		UsageReport:          {MemberlistKV},
 	}
 
+	if t.cfg.InternalServer.Enable {
```

Comment:
what is this code for?

Reply:
This adds the internal server as a dependency of the other services, only it it is enabled.

Reply:
this is kind of tough to parse. can we instead :
- make the Server always dependent on InternalServer
- in initInternalServer make the choice to start it or not
  - if it's disabled you can just return a dummy service like this: https://github.com/grafana/tempo/blob/main/cmd/tempo/app/modules.go#L181